ADBDEV-4911-90: Remove redundant parent check for NULL.

RekGRpth opened the pull request:
Remove redundant parent check for NULL.

In function create_scan_plan pointer best_path->parent cannot be NULL, because
first rel = best_path->parent and a little later scan_clauses =
rel->baserestrictinfo.

In function create_projection_plan pointer best_path->path.parent cannot be NULL
because it is dereferenced above in the function tlist = build_path_tlist(root,
&best_path->path).

So I removed the redundant check for parent for NULL.

bandetto commented:
Can you provide more context on why `best_path->parent` is not `NULL` for each place you've removed the check?

RekGRpth commented on src/backend/optimizer/plan/createplan.c:
```diff
@@ -514,8 +514,7 @@ create_scan_plan(PlannerInfo *root, Path *best_path)
 										   best_path->locus,
-								best_path->parent ? best_path->parent->relids
-										   : NULL,
+										   best_path->parent->relids,
```
Here `best_path->parent` cannot be NULL, because first `rel = best_path->parent` and a little later `scan_clauses = rel->baserestrictinfo`.

RekGRpth commented on src/backend/optimizer/plan/createplan.c:
```diff
@@ -1319,7 +1317,7 @@ create_projection_plan(PlannerInfo *root, ProjectionPath *best_path)
 		plan->flow = cdbpathtoplan_create_flow(root,
 											   best_path->path.locus,
-											   best_path->path.parent ? best_path->path.parent->relids : NULL,
+											   best_path->path.parent->relids,
```
Here `best_path->path.parent` cannot be NULL because it is dereferenced above in the function `tlist = build_path_tlist(root, &best_path->path)`.

RekGRpth committed:
Remove redundant parent check for NULL.

In function create_scan_plan pointer best_path->parent cannot be NULL, because
first rel = best_path->parent and a little later scan_clauses =
rel->baserestrictinfo.

In function create_projection_plan pointer best_path->path.parent cannot be NULL
because it is dereferenced above in the function tlist = build_path_tlist(root,
&best_path->path).

So I removed the redundant check for parent for NULL.
